File: JavaScript_Courses/JavaScript_Basics/Preparation/03_Conditional_Statements_Advanced_Exercise/06_Operations_Between_Numbers.js
function operationsBetweenNumbers(input) {
    let num1 = Number(input[0]);
    let num2 = Number(input[1]);
    let operator = input[2];

    let result = 0;
    let result_as_string;

    if (operator == "+") {
        result = num1 + num2;
        let evenOrOdd;

        if (result % 2 == 0) {
            evenOrOdd = "even";
        } else {
            evenOrOdd = "odd";
        }

        result_as_string = `${num1} ${operator} ${num2} = ${result} - ${evenOrOdd}`;

    } else if (operator == "-") {
        result = num1 - num2;
        let evenOrOdd;

        if (result % 2 == 0) {
            evenOrOdd = "even";
        } else {
            evenOrOdd = "odd";
        }

        result_as_string = `${num1} ${operator} ${num2} = ${result} - ${evenOrOdd}`;

    } else if (operator == "*") {
        result = num1 * num2;
        let evenOrOdd;

        if (result % 2 == 0) {
            evenOrOdd = "even";
        } else {
            evenOrOdd = "odd";
        }

        result_as_string = `${num1} ${operator} ${num2} = ${result} - ${evenOrOdd}`;
        
    } else if (operator == "/") {
        if (num2 == 0) {
            result_as_string = `Cannot divide ${num1} by zero`;
        } else {
            result = num1 / num2;
            result_as_string = `${num1} ${operator} ${num2} = ${result.toFixed(2)}`;
        }

    } else if (operator == "%") {
        if (num2 == 0) {
            result_as_string = `Cannot divide ${num1} by zero`;
        } else {
            result = num1 % num2;
            result_as_string = `${num1} ${operator} ${num2} = ${result}`;
        }
        
    }

    console.log(result_as_string);
}

// operationsBetweenNumbers(["10", "12", "+"]);

File: Python_Courses/Python_Basics/Python_PB_2025_03/03_Conditional_Statements_Advanced_Exercise/06_Operations_Between_Numbers.py
# : Събиране(+), Изваждане(-), Умножение(*),  Деление(/) и Модулно деление(%).

n1 = int(input())
n2 = int(input())
operator_string = input()  # "+", "-", "*", "/", or "%"

result = ''
if operator_string == '+':
    math_result = n1 + n2
    even_or_odd = ''
    if math_result % 2 == 0:
        even_or_odd = 'even'
    else:
        even_or_odd = 'odd'

    result = f'{n1} {operator_string} {n2} = {math_result} - {even_or_odd}'
    # result = f'{n1} {operator_string} {n2} = {math_result} - {"even" if math_result % 2 == 0 else "odd"}'

elif operator_string == '-':
    math_result = n1 - n2
    even_or_odd = ''
    if math_result % 2 == 0:
        even_or_odd = 'even'
    else:
        even_or_odd = 'odd'

    result = f'{n1} {operator_string} {n2} = {math_result} - {even_or_odd}'

elif operator_string == '*':
    math_result = n1 * n2
    even_or_odd = ''
    if math_result % 2 == 0:
        even_or_odd = 'even'
    else:
        even_or_odd = 'odd'

    result = f'{n1} {operator_string} {n2} = {math_result} - {even_or_odd}'

elif operator_string == '/':
    if n2 == 0:
        result = f'Cannot divide {n1} by zero'
    else:
        math_result = n1 / n2
        result = f'{n1} / {n2} = {math_result:.2f}'

elif operator_string == '%':
    if n2 == 0:
        result = f'Cannot divide {n1} by zero'
    else:
        math_result = n1 % n2
        result = f'{n1} % {n2} = {math_result}'

print(result)
